### ISSUE TITLE
`ydotool mousemove --absolute`  moves Mouse to the top of the Screen

### ISSUE THREAD

**Issue by JakobDev** [ReimuNotMoe/ydotool#250]
`ydotool mousemove --absolute` just moves the Mouse to the upper left Corner of the Screen, no matter which coordinates are entered.

**Comment by mashrit** [comment]
any progress on this matter?
i also have this issue, so i need to move the mouse to the top left corner of the screen and then use the relative function to move it to the desired location
it is really annoying

**Comment by jonas73x** [comment]
did you see this comment in the help? 
"You need to disable mouse speed acceleration for correct absolute movement."
does it make a difference?

**Comment by mashrit** [comment]
yes pointer acceleration is turned off in the settings

**Comment by M4rtineski** [comment]
Disabling pointer acceleration helped. Thanks jonas! I've read that I needed to do that but it slipped my mind still.

![image](https://github.com/user-attachments/assets/e0eba637-40a0-470b-a9d7-53d2b6604309)

Edit: Oh, but it seems to be inconsistent at it though. Sometimes it will go to specified location and sometimes it will go to the corner...

Edit2: I think I've figured this out? When I try multiple relative mousemove commands at once to achieve the same effect as when using 'absolute' argument I'm experiencing a very similar issue where the commands work randomly. When trying them one by one repeatedly using separate shortcuts for each command they worked flawlessly. So depending on how this 'absolute' argument is handled in the backend a small pause between the actions may do the trick. 

Edit3: Something like `ydotool mousemove -x -3000 -y -3000 && sleep 0.05 && ydotool mousemove -x 1360 -y 425` doesn't cause me problems.

**Comment by cen1** [comment]
I don't have an option to turn off mouse accel in Debian KDE, I only get "Flat" and "Adaptive" in Mouse settings.

A workaround which works is to set absolute to 0,0 to move cursor to 0,0 and then use relative to achieve the desired absolute position. Works for my use case.

**Comment by davidlevner** [comment]
I have the same problem. I am using KDE with Fedora 41. I turned off mouse acceleration but it didn't help. I am using the same workaround as mashrit.

**Comment by roadbloack** [comment]
@davidlevner 
I was so annoyed by this issue, I looked for an alternative and found [this](https://github.com/tio/input-emulator), you can use touch for absolute movement and mouse for relative

**Comment by Mte90** [comment]
Same issue but disabling pointer accelration didn't fixed.
In my case I have a multi monitor system and doesn't work very good

**Comment by garbofrogman** [comment]
I'm having the same problem
- Fedora 41
- KDE Plasma 6.3.1
- KWin (Wayland)
- 2 monitors

Turning off mouse accelleration did not fix it

**Comment by CBx86** [comment]
Same

FC41
KDE Plasma 6.3.3
Kwin
1 Monitor x 2 Workspaces
0ff mouse acceleration dont fix.

Thanks!

**Comment by CBx86** [comment]
Hey guys, some news? Any alternative software?

Thanks!

**Comment by Renari** [comment]
The easiest thing to do is just do `ydotool mousemove -x -9999999 -y -9999999` a couple times to force the mouse to 0,0. Then you can have your next move be an absolute location via just using the relative command (that is relative from 0,0 is an absolute position).

**Comment by ClassicOldSong** [comment]
ydotool already does this internally, but it seems move to the negative value only once isn't enough.

**Comment by Renari** [comment]
Yeah I noticed this the first time moving to a negative position takes me to around 100,100 however a 2nd call correctly gets you to 0,0.

**Comment by Renari** [comment]
for reference:
```bash
❯ sudo -b ydotoold --socket-path="/tmp/.ydotool_socket" --socket-own="$(id -u):$(id -g)"
Socket path: /tmp/.ydotool_socket
Socket permission: 0600
Socket ownership: UID=1000, GID=1000
READY
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
unable to find device pointer:ydotoold virtual device
❯ YDOTOOL_SOCKET="/tmp/.ydotool_socket" ydotool mousemove --absolute -x 100 -y 100
❯ kdotool getmouselocation
x:39 y:31 screen:3 window:{846c200e-866f-4b7f-b423-f4f903135311}
❯ YDOTOOL_SOCKET="/tmp/.ydotool_socket" ydotool mousemove --absolute -x 100 -y 100
❯ kdotool getmouselocation
x:0 y:0 screen:3 window:{e8d01825-e1e3-4e02-93c2-990e76207b4c}
```

This is on:
KDE Plasma Version: 6.5.3
KDE Frameworks Version: 6.20.0

**Comment by amariusz** [comment]
EDIT: this comment applies to X11 session. See explanation below.

In my case, I managed to fix `--absolute` mode behavior by disabling the "Screen edges" action for the top-left corner in KDE settings — set to "No Action".

Now:
> ydotool mousemove --absolute -x 100 -y 100

works correctly.

By the way, an interesting observation — when an action is assigned to the corner:

> $ ydotool mousemove -x -99999 -y -99999
> $ kdotool getmouselocation --shell
> > X=1
> > Y=1

when it's set to No Action:

> $ ydotool mousemove -x -99999 -y -99999
> $ kdotool getmouselocation --shell
> > X=0
> > Y=0

It seems that KDE doesn't allow ydotool to set the cursor position to (0,0) and instead bounces it to (1,1), which might be the cause of this odd behavior.

Also worth mentioning — it's a good idea to disable "Enable pointer acceleration" in the Mouse settings. Not because it's strictly required for `--absolute` mode, but because even relative movement ends up landing at a different position than requested.

> Operating System: Fedora Linux 44
> KDE Plasma Version: 6.7.4
> ydotool.x86_64 1.0.4-8.fc44
X11 (plasma-workspace-x11)

**Summary:** it's worth trying to disable the active screen corner and mouse pointer acceleration.

**Comment by Renari** [comment]
I tried this and it didn't work out for me, I already had screen edges disabled:
```
❯ YDOTOOL_SOCKET="/tmp/.ydotool_socket" ydotool mousemove --absolute -x 0 -y 0
❯ kdotool getmouselocation
x:48 y:14 screen:0 window:{55a8aa3d-eea6-45d3-a1d5-5f65bbf6e398}
❯ YDOTOOL_SOCKET="/tmp/.ydotool_socket" ydotool mousemove --absolute -x -999999 -y -999999
❯ kdotool getmouselocation
x:39 y:31 screen:0 window:{55a8aa3d-eea6-45d3-a1d5-5f65bbf6e398}
```
The move location isn't even consistent, it's like mouse acceleration is on, but I've triple checked all of my mouse devices in plasma settings (including the ydotool one) and all of them have pointer acceleration off.

**Comment by amariusz** [comment]
I thought about it more and realized I'm running on an X11 session in Plasma 6.7. On Wayland, among other things, the clipboard in the Deskflow app didn't work for me, so as a workaround I installed the `plasma-workspace-x11` package, and as a result I also tested ydotool on that environment.

Indeed, after logging into a Wayland session I'm observing very similar behavior to what you described — `--absolute` lands at different points, just as you wrote, as if acceleration were enabled. I made sure it's disabled for the `ydotool virtual device`, and thanks to that at least relative navigation works.

So at this point, the safe common denominator for both sessions is relative movement. Out of curiosity, I added session detection to my script using `XDG_SESSION_TYPE`, and on X11 I use `--absolute`, while on Wayland I use relative. I'm not sure if that actually changes anything in practice, since in both cases ydotool flickers the cursor in the top-left corner.

**Comment by amariusz** [comment]
I've come to the conclusion that the simplest way to achieve consistent behavior supporting absolute-type navigation without the cursor blinking in the corner is to wrap kdotool and ydotool together.

In its simplest form, such a wrapper script `ydotool_mousemove_absolute.sh X Y` would be 3 lines:

```
#!/usr/bin/env bash
eval "$(kdotool getmouselocation --shell)"
ydotool mousemove -x "$(( $1 - X ))" -y "$(( $2 - Y ))"
```
